plugins/bufferlines/barbar.nix

```diff
@@ -1,58 +1,53 @@
 { lib, pkgs, config, ... }:
+
 with lib;
+
 let
-  cfg = config.programs.nixvim.plugins.barbar;
-in
-{
-  options.programs.nixvim.plugins.barbar = {
-    enable = mkEnableOption "Enable barbar.nvim";
-
-    animations = mkOption {
-      type = types.nullOr types.bool;
-      default = null;
-      description = "Enable animations";
-    };
-
-    autoHide = mkOption {
-      type = types.nullOr types.bool;
-      default = null;
-      description = "Auto-hide the tab bar when there is only one buffer";
-    };
-
-    closeable = mkOption {
-      type = types.nullOr types.bool;
-      default = null;
-      description = "Enable the close button";
-    };
-
-    icons = mkOption {
-      type = with types; nullOr (oneOf [bool (enum ["numbers both"])]);
-      default = null;
-      description = "Enable/disable icons";
-    };
-
-    iconCustomColors = mkOption {
-      type = with types; nullOr (oneOf [bool str]);
-      default = null;
-      description = "Sets the icon highlight group";
-    };
-
-    # Keybinds concept:
-    # keys = {
-    #   previousBuffer = mkBindDef "normal" "Previous buffer" { action = ":BufferPrevious<CR>"; silent = true; } "<A-,>";
-    #   nextBuffer = mkBindDef "normal" "Next buffer" { action = ":BufferNext<CR>"; silent = true; } "<A-.>";
-    #   movePrevious = mkBindDef "normal" "Re-order to previous" { action = ":BufferMovePrevious<CR>"; silent = true; } "<A-<>";
-    #   moveNext = mkBindDef "normal" "Re-order to next" { action = ":BufferMoveNext<CR>"; silent = true; } "<A->>";
-
-    #   # TODO all the other ones.....
-    # };
-  };
 
-  config.programs.nixvim = mkIf cfg.enable {
-    extraPlugins = with pkgs.vimPlugins; [
-      barbar-nvim nvim-web-devicons
-    ];
+  name = "barbar";
+
+  helpers = import ../helpers.nix { inherit lib config; };
+  cfg = config.programs.nixvim.plugins.${name};
 
-    # maps = genMaps cfg.keys;
+  moduleOptions = with helpers; {
+    animation = boolOption true "Enable animations";
+    autoHide = boolOption false "Auto-hide the tab bar when there is only one buffer";
+    tabpages = boolOption true "Enable 'current/total' tabpages indicator in top right corner";
+    closable = boolOption true "Enable the close button";
+    clickable = boolOption true "Enable clickable tabs\n - left-click: go to buffer\n - middle-click: delete buffer";
   };
+
+  # pluginOptions = {
```

Comment:
I'd delete this commented-out code

Reply:
Sure, was from before I added the `toLuaOptions` helper